httpcache.go

```diff
@@ -498,6 +506,44 @@ func headerAllCommaSepValues(headers http.Header, name string) []string {
 	return vals
 }
 
+type cachedReadCloser struct {
+	resp     *http.Response
+	origBody io.ReadCloser
```

Comment:
Also, given it's being assigned just `resp.Body` in `newCachedReadCloser`, or could we always just use `resp.Body`? If not, why not?